### PR TITLE
fix(44701): Rename Symbol(F2) failed to rename the rhs of "===" when operate on union of string literal

### LINKED ISSUE #44701: Rename Symbol(F2) failed to rename the rhs of "===" when operate on union of string literal
# Bug Report

### 🔎 Search Terms

rename symbol string literal union

### 🕗 Version & Regression Information

- This is the behavior in every version I tried

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.4.0-dev.20210622#code/C4TwDgpgBAysCGxoF4oHIB2B7YBaAThPACYhpQA+6hJZA3AFAD0TUb7HnX3Aes690FCoYAK7AowABbQAxqPwBnLPigzCAGhGFFiqADEATFHgZiUQhngBbaAAMapO5KxQ7WANZ2AdA1kAbeF0oAEEoAG8GDkUEJAAuWFiUdGw8R3oGKPZRMGJECAAKAEoIrK4ASwAzKALpcsVvGPyoZFbqIlI0EsjhNjLOAF9+oYGgA)

### 💻 Code

```ts
type State = 'not-ready' | 'ready';
//                            ^
//                          put the cursor here, press F2 and rename `ready` to `ok`.
class A {
    state: State = 'not-ready';

    update() {
        if (this.state === 'ready') {
            
        }
    }
}
```

### 🙁 Actual behavior

```ts
type State = 'not-ready' | 'ok';
class A {
    state: State = 'not-ready';

    update() {
        if (this.state === 'ready') { 
        // This condition will always return 'false' since the types 'State' and '"ready"' have no overlap.(2367)
            
        }
    }
}
```

### 🙂 Expected behavior

```ts
type State = 'not-ready' | 'ok';
class A {
    state: State = 'not-ready';

    update() {
        if (this.state === 'ok') {
            
        }
    }
}
```

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #44701

### CODE REVIEW

File: tests/cases/fourslash/renameStringLiteralOk1.ts [hunk 1]
```diff
@@ -9,10 +9,3 @@
 //// ff.f = '[|foo|]'
 
 verify.rangesWithSameTextAreRenameLocations("foo");
-
```
Comment (Kingwl) [inline tests/cases/fourslash/renameStringLiteralOk1.ts:12]: Oops. My bad.

Comment (a-tarasyuk) [reply tests/cases/fourslash/renameStringLiteralOk1.ts:12]: ;)

**Comment by whzx5byb** [comment]
The `switch` statement also has the same issue with `===`. 
Would you please fix them in this PR? Thank you.

**Comment by a-tarasyuk** [comment]
@whzx5byb I think you can create new issue related to `switch` statement, with detailed Actual/Expected behavior

File: src/services/utilities.ts [hunk 2]
```diff
@@ -800,6 +800,11 @@ namespace ts {
             return contextualType;
         }
 
+        const parent = node.parent;
```
Comment (sandersn) [inline src/services/utilities.ts:803]: Is it all safe to put this new code into getContextualType? What happens if you put it there?

**Review by sandersn — CHANGES_REQUESTED** [review]
Should this be part of the compiler instead of adding it into services afterward?

**Comment by sandersn** [comment]
Hmmm, those changes from `T expands "a" | "b"` to `"a" | "b"` are worrisome. I wonder if that's happening because of narrowing? But that wouldn't really make sense either.

**Comment by a-tarasyuk** [comment]
@sandersn You are right. I've reverted changes.

**Review by sandersn — APPROVED** [review]
Let's take this for now, but someday I'd like to learn why this works for services but not for the checker.

**Pull request merged**